WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -47,26 +49,29 @@ public async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo in
 		var loadingRing = new LoadingSuggestionViewModel();
 		Suggestions.Add(loadingRing);
 
-		var hasChange = transaction.InnerWalletOutputs.Any(x => x.ScriptPubKey != destination.ScriptPubKey);
-
-		if (hasChange && !isFixedAmount)
+		try
 		{
-			try
+			var hasChange = transaction.InnerWalletOutputs.Any(x => x.ScriptPubKey != destination.ScriptPubKey);
+
+			if (hasChange && !isFixedAmount)
 			{
 				var suggestions =
-					ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, destination, wallet, cancellationToken);
+					ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, destination, wallet, cancellationTokenSource.Token);
 
 				await foreach (var suggestion in suggestions)
 				{
 					Suggestions.Insert(Suggestions.Count - 1, suggestion);
 				}
 			}
-			catch (OperationCanceledException)
-			{
-				Logger.LogWarning("Suggestion creation has timed out.");
-			}
 		}
-
-		Suggestions.Remove(loadingRing);
+		catch (OperationCanceledException)
+		{
+			Logger.LogWarning("Computing privacy suggestions timed out.");
+		}
+		finally
+		{
+			cancellationTokenSource.Cancel();
```

Comment:
You should not cancel too

Reply:
Oh, thanks 😅. I didn't know disposing CTS will cancel it also.